custom-modifiers/waitforimagemodifier/src/main/java/WaitForImageModifier.java

```diff
@@ -0,0 +1,85 @@
+import com.cognifide.aet.communication.api.metadata.CollectorStepResult;
+import com.cognifide.aet.job.api.collector.CollectorJob;
+import com.cognifide.aet.job.api.exceptions.ParametersException;
+import com.cognifide.aet.job.api.exceptions.ProcessingException;
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.concurrent.TimeUnit;
+
+public class WaitForImageModifier implements CollectorJob {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WaitForImageModifier.class);
+
+    static final String NAME = "wait-for-image";
+
+    private final WebDriver webDriver;
+
+    private static final String CSS_PARAMETER = "css";
+    private static final String XPATH_PARAMETER = "xpath";
+    private static final String TIMEOUT_PARAMETER = "timeout";
+
+    private String webElementCssLocator;
+    private String webElementXpathLocator;
+    private int timeoutInMilliseconds;
+
+    WaitForImageModifier(WebDriver webDriver) {
+        this.webDriver = webDriver;
+    }
+
+    @Override
+    public CollectorStepResult collect() throws ProcessingException {
```

Comment:
Those two are pretty cool modifiers. In AET we try to have as much features testing as possible automated.
Again, tutorial for creating AET integration tests is in progress, however you can find ``AET integration-tests module` [here](https://github.com/Cognifide/aet/tree/master/integration-tests).
It contains several sub-modules. The idea here is to create a demo-page in `sample-site` that later can be automatically tested with AET features (like Collectors, Modifiers, Comparators and Data Filters) with sanity [`test-suite`](https://github.com/Cognifide/aet/tree/master/integration-tests/test-suite). It is assembled from `partials`.
See example suite partial for [Click modifier partial](https://github.com/Cognifide/aet/blob/master/integration-tests/test-suite/partials/click.xml).